modules/Logger.js

```diff
@@ -0,0 +1,169 @@
+(function(root, factory) {
+  if (typeof define === 'function' && define.amd) {
+    define(['underscore', 'backbone', './ServiceCell'], factory);
+  } else if (typeof exports === 'object') {
+    module.exports = factory(require('underscore'), require('backbone'), require('./ServiceCell'));
+  } else {
+    root.Torso = root.Torso || {};
+    root.Torso.Logger = factory(root._, root.Backbone, root.Torso.ServiceCell);
+  }
+}(this, function(_, Backbone, ServiceCell) {
+  'use strict';
+
+  var Logger = ServiceCell.extend({ 
+
+    log : {}, 
+    sessions: [],
+    unclosed: {},
+    _isClosed: true,
+    _sessionStart: null,
+    _sessionEnd: null,
+    _sessionID: null,
+
+    /**
+    * saves information about event and wraps start and end signals in 'session' 
+    * @ param {JSON object} contains tracking information (including state: 'start' or 'end', uuid, sessionID)
+    * @ method track
+    */
+  	track: function(eventInfo){
+      if (eventInfo.state){
+        if (this._isClosed){
+          this.resetSession();
+          if (eventInfo.state == "start") {
+            this.openSession(eventInfo);
+            this.startEvent(eventInfo);
+          }
+        } else {
+          if (eventInfo.state == "start") {
+            this.startEvent(eventInfo);
+          }
+          else if (eventInfo.state == "end") {
+              this.endEvent(eventInfo);
+              if (Object.keys(this.unclosed).length === 0) {
+                this.closeSession();
+              }
+          }
+        }
+      }
+  	},
+
+    /**
+    * tracks information about the 'start' of events
+    * @param {JSON object} contains tracking information including UUID and time
+    * @method startEvent
+    */
+    startEvent: function(eventInfo){
+      var uuid = eventInfo.UUID;
```

Comment:
Also, probably best to generate a UUID for people here, and return the UUID at the end of this method.